api/v1beta2/kustomization_types.go

```diff
@@ -0,0 +1,364 @@
+	meta.SetResourceCondition(k, meta.ReadyCondition, status, reason, trimString(message, MaxConditionMessageLength))
+	k.Status.ObservedGeneration = k.Generation
+	k.Status.LastAttemptedRevision = revision
+}
+
+// KustomizationNotReady registers a failed apply attempt of the given Kustomization.
+func KustomizationNotReady(k Kustomization, revision, reason, message string) Kustomization {
+	SetKustomizationReadiness(&k, metav1.ConditionFalse, reason, trimString(message, MaxConditionMessageLength), revision)
+	if revision != "" {
+		k.Status.LastAttemptedRevision = revision
+	}
+	return k
+}
+
+// KustomizationNotReadyInventory registers a failed apply attempt of the given Kustomization.
+func KustomizationNotReadyInventory(k Kustomization, inventory *Inventory, revision, reason, message string) Kustomization {
+	SetKustomizationReadiness(&k, metav1.ConditionFalse, reason, trimString(message, MaxConditionMessageLength), revision)
+	SetKustomizationHealthiness(&k, metav1.ConditionFalse, reason, reason)
+	if revision != "" {
+		k.Status.LastAttemptedRevision = revision
+	}
+	k.Status.Inventory = inventory
+	return k
+}
+
+// KustomizationReadyInventory registers a successful apply attempt of the given Kustomization.
+func KustomizationReadyInventory(k Kustomization, inventory *Inventory, revision, reason, message string) Kustomization {
+	SetKustomizationReadiness(&k, metav1.ConditionTrue, reason, trimString(message, MaxConditionMessageLength), revision)
+	SetKustomizationHealthiness(&k, metav1.ConditionTrue, reason, reason)
+	k.Status.Inventory = inventory
+	k.Status.LastAppliedRevision = revision
+	return k
+}
+
+// GetTimeout returns the timeout with default.
+func (in Kustomization) GetTimeout() time.Duration {
+	duration := in.Spec.Interval.Duration - 30*time.Second
+	if in.Spec.Timeout != nil {
+		duration = in.Spec.Timeout.Duration
+	}
+	if duration < 30*time.Second {
+		return 30 * time.Second
+	}
+	return duration
+}
+
+// GetRetryInterval returns the retry interval
+func (in Kustomization) GetRetryInterval() time.Duration {
+	if in.Spec.RetryInterval != nil {
+		return in.Spec.RetryInterval.Duration
+	}
+	return in.Spec.Interval.Duration
+}
+
+// GetDependsOn returns the list of dependencies across-namespaces.
+func (in Kustomization) GetDependsOn() (types.NamespacedName, []dependency.CrossNamespaceDependencyReference) {
+	return types.NamespacedName{
+		Namespace: in.Namespace,
+		Name:      in.Name,
+	}, in.Spec.DependsOn
+}
+
+// GetStatusConditions returns a pointer to the Status.Conditions slice.
+func (in *Kustomization) GetStatusConditions() *[]metav1.Condition {
+	return &in.Status.Conditions
+}
+
+const (
+	// GitRepositoryIndexKey is the key used for indexing kustomizations based on their Git sources.
+	GitRepositoryIndexKey string = ".metadata.gitRepository"
+	// BucketIndexKey is the key used for indexing kustomizations based on their S3 sources.
+	BucketIndexKey string = ".metadata.bucket"
+)
```

Comment:
Given index keys are client-specific, I think this is reconciler / application logic.